### ISSUE TITLE
Locality aware routing not applied when empty locality config on proxy

### ISSUE THREAD

**Issue by anilutcs** [istio/istio#11609]
**Describe the bug** 
Locality aware config is not applied when side car proxy has empty locality config, something like:
```
locality: 
{
}
```

**Expected behavior**
Locality aware routing should fall back to locality of side car service instances when both i) locality is not set or ii) empty locality is set on side car.  

**Steps to reproduce the bug**
1. Checkout and build istio/release-1.1 branch commit `a0ac8afe5b884b39fe69af5b61f9a2215df8d69f`
2. Enable locality aware routing (Add the following to mesh config):
```
LocalityLbSetting:
  failover:
    - from: us-west-2
         to: us-east-2
```
3. Deploy sleep service with side car injection
4. Deploy the following service entry:
```
apiVersion: networking.istio.io/v1alpha3
kind: ServiceEntry
metadata:
  name: server-e2e-mc
spec:
  hosts:
  - server.mesh.com
  ports:
  - number: 80
    name: http
    protocol: HTTP
  location: MESH_EXTERNAL
  resolution: DNS
  endpoints:
  - address: server.us-west-2.elb.amazonaws.com
    ports:
      http: 443
    locality: us-west-2
  - address: server.us-east-2.elb.amazonaws.com
    ports:
      http: 443
    locality: us-east-2
```
At this point the envoy configuration generated for client's sidecar doesn't have locality failover config for _server.mesh.com_ cluster.

**Version**
Branch: `release-1.1`
Commit: `a0ac8afe5b884b39fe69af5b61f9a2215df8d69f`

**Installation**
Built locally and installed using helm charts

**Environment**
AWS

**Comment by hzxuzhonghu** [comment]
Thanks, I will fix it. It should be easy since we have implemented it in #10720

**Comment by hzxuzhonghu** [comment]
opened #11727 to move it from cluster.go to ads.go.  Which is more appropriate place for it.
